usecases/permissions_requests.go

```diff
@@ -0,0 +1,120 @@
+package usecases
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/topfreegames/Will.IAM/models"
+	"github.com/topfreegames/Will.IAM/repositories"
+)
+
+// PermissionsRequests define entrypoints for PermissionsRequests actions
+type PermissionsRequests interface {
+	Create(*models.PermissionRequest) error
+	Deny(string, string) error
+	Grant(string, string) error
+	ListOpenRequestsVisibleTo(
+		*repositories.ListOptions, string,
+	) ([]models.PermissionRequest, int64, error)
+	WithContext(context.Context) PermissionsRequests
+}
+
+type permissionsRequests struct {
+	repo *repositories.All
+	ctx  context.Context
+}
+
+func (prs permissionsRequests) WithContext(ctx context.Context) PermissionsRequests {
+	return &permissionsRequests{prs.repo.WithContext(ctx), ctx}
+}
+
+// Create checks if pr.saID has open request OR has permission, if not it opens a permission request
+func (prs permissionsRequests) Create(pr *models.PermissionRequest) error {
+	return prs.repo.WithPGTx(prs.ctx, func(repo *repositories.All) error {
+		pr.State = models.PermissionRequestStates.Open
+		has, err := repo.ServiceAccounts.HasPermission(pr.ServiceAccountID, pr.Permission())
+		if err != nil {
+			return err
+		}
+		if has {
+			// TODO: replace by proper error
```

Comment:
What's a proper error? Why not fix this now?

Reply:
I'll come back to errors after this PR. My idea is to reorganize pkg errors. Most places return 500 when we could return 4xx.

Reply:
I'd suggest adding your username to the TODO and rephrasing as:

`// TODO(ghostec): Revisit returned error codes`

so it's clear to others what it's about.

Reply:
Done